app/Http/Controllers/ProjectController.php

```diff
@@ -162,50 +162,69 @@ public function projectUserSearch(Request $request): array
 
     public function index(): Response|ResponseFactory
     {
-        $projects = Project::query()
-            ->with([
-                'checklists.tasks.checklist.project',
-                'access_budget',
-                'categories',
-                'comments.user',
-                'departments.users.departments',
-                'genres',
-                'managerUsers',
-                'project_files',
-                'sectors',
-                'users.departments',
-                'writeUsers',
-                'state',
-                'delete_permission_users'
-            ])
+        $entitiesPerPage = request()->get('entitiesPerPage', 10);
+
+        $projectsQuery = Project::with([
+            'access_budget' => function ($query): void {
+                $query->without(['calendar_settings', 'calendarAbo', 'shiftCalendarAbo', 'vacations']);
+            },
+            'categories',
+            'genres',
+            'managerUsers' => function ($query): void {
+                $query->without(['calendar_settings', 'calendarAbo', 'shiftCalendarAbo', 'vacations']);
+            },
+            'users' => function ($query): void {
+                $query->without(['calendar_settings', 'calendarAbo', 'shiftCalendarAbo', 'vacations']);
+            },
+            'writeUsers' => function ($query): void {
+                $query->without(['calendar_settings', 'calendarAbo', 'shiftCalendarAbo', 'vacations']);
+            },
+            'state',
+            'delete_permission_users' => function ($query): void {
+                $query->without(['calendar_settings', 'calendarAbo', 'shiftCalendarAbo', 'vacations']);
+            },
+        ])->whereNull('pinned_by_users')
             ->orderBy('id', 'DESC')
-            ->get();
+            ->without(['shiftRelevantEventTypes']);
+
+        if (request()->has('search')) {
+            $projectsQuery = Project::search(request()->get('search'));
+        }
+        $users = collect();
+        if (request()->has('user_search') && request()->get('user_search') !== null && request()->get('user_search') !== '') {
+            $users = User::search(request()->get('user_search'))->get();
+        }
+
+        $projects = $projectsQuery->paginate($entitiesPerPage);
 
         return inertia('Projects/ProjectManagement', [
```

Comment:
Funktionalitäten für alle neuen Sachen müssen in Services/Repositories ausgelagert werden. Anschließend brauchen die Service/Repository-Funktionen dann noch Unit-Tests

Reply:
Wurde in einen Service ausgelagert